tools/gen_allocs.c

```diff
@@ -589,30 +589,97 @@ gen_dealloc1 ( char * dirname )
   FILE * fp ;
   char  fname[NAMELEN] ;
   char * fn = "deallocs.inc" ;
+  // Open array of deallocs_[n].inc
+  int   numFiles = 12;
+  int    idx      = 0;
+  FILE * fpSub; 
+  char * filename_prefix = "deallocs_" ;
 
   if ( dirname == NULL ) return(1) ;
   if ( strlen(dirname) > 0 ) { sprintf(fname,"%s/%s",dirname,fn) ; }
   else                       { sprintf(fname,"%s",fn) ; }
   if ((fp = fopen( fname , "w" )) == NULL ) return(1) ;
   print_warning(fp,fname) ;
-  gen_dealloc2( fp , "grid%", &Domain ) ;
+
+  fprintf(
+          fp,
+          "INTERFACE\n"
+          );
+  if ( dirname == NULL ) return(1) ;
+  for ( ; idx < numFiles; idx++ )
+  {
+    if ( strlen(dirname) > 0 ) { sprintf(fname,"%s/%s%d.F90",dirname,filename_prefix,idx) ; }
```

Comment:
Will using `F90` suffixes here lead to problems on systems without case-sensitive filesystems? Do the generated `*.F90` files ever get turned into `*.f90` files?

Reply:
These would rely on implicit `.o : .F90` rules, such as the one that was added for the DA build. 

It seems like the problem, as noted in #2057, is usually from clobbering the original source files. As these are auto-generated from the registry overwriting them should not be an issue.